baseline/config/run_training.py

```diff
@@ -0,0 +1,174 @@
+                    bounds[1] = 1e+100
+
+                params += [CSH.UniformFloatHyperparameter(paramname, float(bounds[0]), float(bounds[1]), default_value=float(default))]
+            
+            elif paramtype == 'integer':
+                params += [CSH.UniformIntegerHyperparameter(paramname, lower=int(bounds[0]), upper=int(bounds[1]), default_value=int(default))]
+        i += 1
+
+    return params
+
+# runs ecole and returns primal-dual integral as a reward
+def runEcole(settings, instance):
+
+    print("New Ecole run with instance ", instance)
+
+    # start a new episode
+    env.reset(instance)
+
+    # get the next action from the given settings
+    action = {k: settings[k] for k in settings}
+
+    for k in action.keys():
+        if action[k] == 'TRUE':
+            action[k] = True
+        elif action[k] == 'FALSE':
+            action[k] = False
+
+    # apply the action and collect the reward
+    _, _, reward, _, _ = env.step(action)
+
+    return reward
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        'problem',
+        help='Problem benchmark to process.',
+        choices=['item_placement', 'load_balancing', 'anonymous'],
+    )
+    parser.add_argument(
+        '-t', '--timelimit',
+        help='Instance time limit (in seconds).',
+        default=300,
+        type=float,
+    )
+    parser.add_argument(
+        '-i', '--ninstances',
+        help='Number of training instances.',
+        default=10,
+        type=int,
+    )
+    parser.add_argument(
+        '-e', '--nevaluations',
+        help='Number of function evaluations of SMAC.',
+        default=10,
+        type=int,
+    )
+    args = parser.parse_args()
+
+    # collect the instance files
+    if args.problem == 'item_placement':
+        instances_path = pathlib.Path(f"../../instances/1_item_placement/train/")
+    elif args.problem == 'load_balancing':
+        instances_path = pathlib.Path(f"../../instances/2_load_balancing/train/")
+    elif args.problem == 'anonymous':
+        instances_path = pathlib.Path(f"../../instances/3_anonymous/train/")
+
+    print(f"Processing instances from {instances_path.resolve()}")
+    instance_files = list(instances_path.glob('*.mps.gz'))
+
+    # randomly choose set of training instances
+    instances = random.sample(instance_files, 10)
+
+    # write instances to a file
+    instancefile = open('instances.txt', "a+")
+    for i in range(len(instances)):
+        instancefile.write(str(instances[i]) + '\n')
+    instancefile.close()
+    
+    # create ecole environment
+    env = ec.environment.Configuring(
+          
+          # set time limit for each instance
+          scip_params = {'limits/time' : args.timelimit},
+          
+          # pure bandit, no observation
+          observation_function = None,
+
+          # minimize the primal-dual integral
+          reward_function = ec.reward.PrimalDualIntegral(),
```

Comment:
Ideally the training reward should be the same primal-dual integral that is used for evaluation (this is not a requirement as participants can do whatever you want for training, but for a baseline / example implementation it might be less confusing to do so). The reward can be found in `common/rewards.py`. To import this file it can either be copy-pasted here, or messing a bit with `sys.path()` could do as well, similarly to what is done in
https://github.com/ds4dm/ml4co-competition/blob/main/common/evaluate.py#L66
```python
    import sys
    sys.path.insert(1, str(pathlib.Path("../..")))
    import common.rewards
```
Last, you'll have to read the initial primal / dual bounds of each instance as well to initialize the reward function each time you process a new instance, as is done here
https://github.com/ds4dm/ml4co-competition/blob/103b4cf69cf93995fd17347907ca81531295623b/common/evaluate.py#L109-L121

Reply:
You're right, I somehow missed that the evaluation function does not use the default primal-dual integral of ecole. I changed that and now it would be the same.